### DISCUSSION TITLE
Options for supporting search of TSML data via Wordpress

### DISCUSSION THREAD

**Discussion by k8fish3r** [code4recovery/12-step-meeting-list#1314] [Q&A] [answered]
Wondering, if there are routes which members have used to better support search data saved within Twelve Step Meeting List - via external search:
- Wordpress search / Jetpack search / Relevanssi?

Atm Im using 301 director to setup manual redirects. Any thoughts / options users have found helpful very much more than helpful.

IF

CoDA UK Comms / Web
https://codauk.org

**Comment by anchovie91471** [comment]
I'm not sure I understand the use case. So you are wanting to allow users to save searches for quick access or am I misunderstanding?

**Reply by k8fish3r** [reply]
Hi Anch,

so I can see visitors to the website, searching for TSML meetings via Wordpress search (rather than heading to the pages with TSML showing) and searching via TSML. 

I don't find as standard the results returned from TSML, via WordPress search reasonable quality.

Im looking for a route to manually improve quality of WordPress search (and strong results from TSML).

At present I'm setting up 301 redirects.

I wondered is there a better route which is LESS manual?

Any thoughts very much more than welcome.

E.G. https://codauk.org

Searching for London (a 301 redirect forces the visitor to TSML and presents meetings in the region) = what we would like to happen  

Previously a wordpress search returned only articles with "London" in the title, or body (no meetings in the London region). = not helpful 

Kate F

**Comment by joshreisner** [comment, marked as answer]
hi @k8fish3r i just tested your solution and it is pretty nice! i searched "birmingham" in the website search and i was automatically forwarded to the meeting search within 50 miles of birmingham, UK

typically for most of our users the website search is of limited value, so we recommend they hide it - depending on theme, it can often be removed at Appearance > Customize. or just hide it on the meeting page by creating an alternate menu there, or using CSS (you can use the `body.tsml` selector), eg

```
body.tsml .search-toggle-wrapper { display: none; }
```

or another approach might be to customize the search results page with a banner / button like "did you mean to search for CoDA meetings with {terms}?" that takes people from `/?s=test` to `/meetings?tsml-day=any&tsml-query=test`

**Reply by k8fish3r** [reply]
Will add some script to inject into the search results page to improve range of support offered.
Ty Josh.

**Comment by k8fish3r** [comment]
Wordpress seem to have made adjustments to the core which stops redirecting of wordpress searches - shame. 
Will continue to look for a solution which works with the new updates to Wordpress.
